wodles/azure/azure-logs.py

```diff
@@ -341,16 +313,21 @@ def build_log_analytics_query(offset: str, md5_hash: str):
     dict
         The required body for the requested query.
     """
-    min_str, max_str = get_min_max(service_name="log_analytics", md5_hash=md5_hash, offset=offset)
-    # Using "parse" adds compatibility with "last_dates_files" from previous releases as the format wasn't localized
-    # It also handles any datetime with more than 6 digits for the microseconds value provided by Azure
+    item = orm.get_row(orm.LogAnalytics, md5=md5_hash)
```

Comment:
This possibility was already taken into account but was discarded because its use was not consistent between the different Azure service integrations:
- Both `Log Analytics` and `Graph` would require the function to return `min` and `max` both as `str` and `datetime` objects. The alternative would be that `min` and `max` were returned as `str` and then casted to `datetime` outside the function, similar to what we are doing in the final implementation.
- However, `Storage` does not require the `min` and `max` values to be returned as `str` but only as `datetime` objects. Therefore, the options would have been either a function that returns a structure containing the values both as `str` and as `datetime` objects and then just use the ones you want, or a function that is only used for Log analytics and `Graph`, but not for `Storage`.

In the end it was decided to simply return the row object, which contains the `str` values and work accordingly with this object to simplify the process. This also made it easier to log a custom error message for each type of service.

Reply:
This would be easy to do if the module had a class structure, but the way it is it would be problematic, as you are pointing out, so I think it's best to leave it as it is and take it into account for the future.